File: processor/src/test/resources/approvals/jpa/IsbnAttributeConverter.java
package test.boundary.persistence.jpa;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;
import scenarios.java.Isbn;

import javax.annotation.processing.Generated;

@Generated("com.qualityminds.lazyval.processor.internal.codegen.JpaGenerator")
@Converter(
    autoApply = true
)
public class IsbnAttributeConverter implements AttributeConverter<Isbn, String> {
  public String convertToDatabaseColumn(Isbn type) {
    if(type == null) {
      return null;
    }
    return type.getValue();
  }

  public Isbn convertToEntityAttribute(String dbValue) {
    if(dbValue == null) {
      return null;
    }
    return Isbn.parse(dbValue);
  }
}

File: ksp/src/test/resources/approvals/jpa/IsbnAttributeConverter.kt
package test.boundary.persistence.jpa

import jakarta.`annotation`.Generated
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import kotlin.String
import scenarios.kotlin.Isbn

@Generated("com.qualityminds.lazyval.ksp.internal.codegen.JpaGenerator")
@Converter(autoApply = true)
public class IsbnAttributeConverter : AttributeConverter<Isbn?, String?> {
  override fun convertToDatabaseColumn(type: Isbn?): String? = type?.value

  override fun convertToEntityAttribute(dbValue: String?): Isbn? = dbValue?.let { Isbn.parse(it) }
}
